src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts

```diff
@@ -1,61 +1,94 @@
 import { html } from "lit-html";
 import { asyncReplace } from "lit-html/directives/async-replace.js";
+import { createRef, ref, Ref } from "lit-html/directives/ref.js";
+import type QrCreator from "qr-creator"; // XXX: import to only import the _type_ to avoid pulling in the whole module (module itself is used as a dynamic import)
 import {
   DeviceData,
   Timestamp,
 } from "../../../../generated/internet_identity_types";
+import { checkmarkIcon, copyIcon } from "../../../components/icons";
 import { mainWindow } from "../../../components/mainWindow";
+import { toast } from "../../../components/toast";
 import { addDeviceLink } from "../../../utils/addDeviceLink";
 import { AsyncCountdown } from "../../../utils/countdown";
 import { AuthenticatedConnection } from "../../../utils/iiConnection";
-import { renderPage } from "../../../utils/lit-html";
+import { mount, renderPage, withRef } from "../../../utils/lit-html";
 import { delayMillis } from "../../../utils/utils";
 
 const pollForTentativeDeviceTemplate = ({
+  userNumber,
   cancel,
   remaining,
-  link,
+  origin,
 }: {
+  userNumber: bigint;
   cancel: () => void;
   remaining: AsyncIterable<string>;
-  link: string;
+  origin: string;
 }) => {
+  const link = addDeviceLink({ userNumber, origin });
+  const copyLink_ = () => navigator.clipboard.writeText(link);
+
+  const linkCopyElement: Ref<HTMLElement> = createRef();
+  // Copy the link and give visual feedback on success
+  const copyLink = async () => {
+    try {
+      await copyLink_();
+      withRef(linkCopyElement, (linkCopyElement) => {
+        linkCopyElement.classList.add("is-copied");
+      });
+    } catch (e: unknown) {
+      toast.error("Could not copy link to clipboard");
+      console.error("Could not copy link to clipboard", e);
+    }
+  };
+
   const pageContentSlot = html`
     <hgroup>
-      <h1 class="t-title t-title--main">Add a Trusted Device</h1>
-      <p class="t-lead">
-        Complete the steps below
-        <strong class="t-string"> on the device you want to add: </strong>
-      </p>
+      <h2 class="t-paragraph">Identity Anchor ${userNumber}</h2>
+      <h1 class="t-title t-title--main l-stack--none">Add Trusted Device</h1>
+      <p class="t-lead">Follow these steps on your new device</p>
     </hgroup>
+    <div
+      class="t-centered c-qrcode l-stack"
+      ${mount((container) =>
+        container instanceof HTMLElement
+          ? displayQR({ link, container })
+          : undefined
+      )}
+    ></div>
+    <div class="t-wrap c-input c-input--with-button c-input--centered t-weak">
+      <span data-role="add-device-link">${link}</span>
+      <button
```

Comment:
do we have the copy button elsewhere in the code? worth a own component?

Reply:
We have another one, i.e. 2 in total. My rule of thumb is to extract on the 3rd occurrence

Reply:
Good to know, thanks for the feedback